dcap-ql/src/quote.rs

```diff
@@ -379,4 +579,52 @@ mod tests {
         assert_eq!(cd.pcesvn, EXPECTED_PCESVN);
         assert_eq!(cd.pceid, EXPECTED_PCEID);
     }
+
+    pub struct MyVerifier{}
+    impl Quote3SignatureEcdsaP256Verifier for MyVerifier {
+        fn verify_certification_data<'a>(&mut self, quote3signature: &'a Quote3SignatureEcdsaP256) -> Result<Cow<'a, [u8]>> {
+           let cert_chain = quote3signature.certification_data::<Qe3CertDataPckCertChain>()?;
+            Ok(cert_chain.certs[0].clone())
+        }
+
+        /// Verify the quoting enclave (QE3).
+        fn verify_qe3(&mut self, _qe3_report: &[u8], _authentication_data: &[u8]) -> Result<()> {
+            Ok(())
+        }
+    }
+
+    #[test]
+    fn test_quote_verification() {
+        const TEST_QUOTE: &[u8] = &*include_bytes!("../tests/quote_pck_cert_chain.bin");
+        let quote = Quote::parse(TEST_QUOTE).unwrap();
+        let &QuoteHeader::V3 {
+            attestation_key_type,
+            ref qe3_vendor_id,
+            ..
+        } = quote.header();
+
+        assert_eq!(qe3_vendor_id, &&QE3_VENDOR_ID_INTEL[..]);
+
+        assert_eq!(attestation_key_type, Quote3AttestationKeyType::EcdsaP256);
+        let sig = quote.signature::<Quote3SignatureEcdsaP256>().unwrap();
+        #[cfg(feature = "verify")]
+        assert!(Quote3Signature::verify(&sig, TEST_QUOTE).is_ok());
```

Comment:
This check is included in `Quote::verify::<Quote3SignatureEcdsaP256>` below right?

Reply:
Yes, it was old test data that got left behind.